nav2_bt_navigator/behavior_trees/navigate_to_pose_w_replanning_and_recovery.xml

```diff
@@ -11,26 +11,38 @@
         <RateController hz="1.0">
           <RecoveryNode number_of_retries="1" name="ComputePathToPose">
             <ComputePathToPose goal="{goal}" path="{path}" planner_id="GridBased" error_code_id="{compute_path_error_code}"/>
-            <ClearEntireCostmap name="ClearGlobalCostmap-Context" service_name="global_costmap/clear_entirely_global_costmap"/>
+            <Sequence>
+              <WouldAPlannerRecoveryHelp current_error_code="{compute_path_error_code}"/>
+              <ClearEntireCostmap name="ClearGlobalCostmap-Context" service_name="global_costmap/clear_entirely_global_costmap"/>
+            </Sequence>
           </RecoveryNode>
         </RateController>
         <RecoveryNode number_of_retries="1" name="FollowPath">
           <FollowPath path="{path}" controller_id="FollowPath" error_code_id="{follow_path_error_code}"/>
-          <ClearEntireCostmap name="ClearLocalCostmap-Context" service_name="local_costmap/clear_entirely_local_costmap"/>
+          <Sequence>
+            <WouldAControllerRecoveryHelp current_error_code="{follow_path_error_code}"/>
+            <ClearEntireCostmap name="ClearLocalCostmap-Context" service_name="local_costmap/clear_entirely_local_costmap"/>
+          </Sequence>
         </RecoveryNode>
       </PipelineSequence>
-      <ReactiveFallback name="RecoveryFallback">
-        <GoalUpdated/>
-        <RoundRobin name="RecoveryActions">
-          <Sequence name="ClearingActions">
-            <ClearEntireCostmap name="ClearLocalCostmap-Subtree" service_name="local_costmap/clear_entirely_local_costmap"/>
-            <ClearEntireCostmap name="ClearGlobalCostmap-Subtree" service_name="global_costmap/clear_entirely_global_costmap"/>
-          </Sequence>
-          <Spin spin_dist="1.57"/>
-          <Wait wait_duration="5"/>
-          <BackUp backup_dist="0.30" backup_speed="0.05"/>
-        </RoundRobin>
-      </ReactiveFallback>
+      <Sequence>
+        <ReactiveFallback>
+          <WouldAPlannerRecoveryHelp current_error_code="{compute_path_error_code}"/>
```

Comment:
I'm not sure I understand this part - if we're doing global level recoveries, wouldn't we want to do for other reasons? I suppose the one we have right now (spin, backup, clear costmap) wouldn't help potentially though. I'm a little torn on this, whether it should be future-proofed for additional recoveries like teleop or changing goal parameters or the actuality of what's in the tree

Reply:
Ditto here still - what do you think? For this global "break glass in case of emergency" should we just bypass?

Reply:
I don't have a strong opinion either way. I put that in there initially to copy what the current behavior tree behavior. I think if we have recoveries related to global planner errors, a separate sub tree should be built to deal with those errors. If someone wanted to add in teleop they would add a seperate sub recovery tree to check for the errors that interest them.

I think we shouldn't provide default behavior that runs recoveries in a situation that won't help resolve issues which is what you pointed out above. If we find out these recoveries do help resolve global planner issues than maybe look at restructuring the recovery subtree. 

Once we get some users running the new behavior tree we will have a better sense of what direction to take the default tree.